tests/integration/test_integration.py

```diff
@@ -326,15 +300,23 @@ async def initialize_vault_leader(ops_test: OpsTest, app_name: str) -> Tuple[str
     assert leader
     leader_ip = leader.public_address
     vault_url = f"https://{leader_ip}:8200"
-    action_output = await run_get_ca_certificate_action(ops_test)
-    ca_certificate = action_output["ca-certificate"]
-    assert ca_certificate
-    ca_file_location = str(ops_test.tmp_path / "ca_file.txt")
-    with open(ca_file_location, mode="w+") as ca_file:
-        ca_file.write(ca_certificate)
-    vault = Vault(url=vault_url, ca_file_location=ca_file_location)
-    assert not vault.is_initialized()
-    root_token, key = vault.initialize()
+
+    vault = Vault(
+        url=vault_url, ca_file_location=await get_ca_cert_file_location(ops_test, app_name)
+    )
+    if not vault.is_initialized():
+        root_token, key = vault.initialize()
+        await ops_test.model.add_secret(
+            f"root-token-key-{app_name}", [f"root-token={root_token}", f"key={key}"]
+        )
+        return root_token, key
+
+    secrets = await ops_test.model.list_secrets(show_secrets=True)
+    secret = next(secret for secret in secrets if secret.label == f"root-token-key-{app_name}")
+    from base64 import b64decode
```

Comment:
Is there a reason we do the import here dynamically?

Reply:
Yes, but not a good one (at least, not a good reason for it _still_ to be here).

I meant to factor this out to a separate function, since it's a hacky way to get a secret. I left the import with the code.

I need to fix this, thanks for pointing it out.

Reply:
https://github.com/canonical/vault-operator/pull/271/commits/353661c7b80689a26e6abb332df4ea8f10ec4c60